src/process-single-dataset/index.js

```diff
@@ -42,6 +42,18 @@ const processSingleDataset = async (
     datasetJson.featuresDataOrder.indexOf(defaultGroupBy);
 
   const featureDefsData = await readFeatureData();
+  const featuresDataOrder = datasetJson.featuresDataOrder;
+  if (featuresDataOrder.length > featureDefsData.length) {
+    console.error("Error: Too many features in featuresDataOrder");
```

Comment:
this should say `Error: featureDefs has ${featureDefsData.length} features but there are ${featuresDataOrder.length} listed in featuresDataOrder`

Reply:
Sure I'll make changes of them, thank you for the suggestions on error messages! I have created a new branch and moved the code over to it for the key validation issue.  

This PR should be ready to go if we don't have anymore changes on add links data.